File: training-data/python/word_count.py
import collections
import re


def word_count(text):
    words = re.findall(r"[a-zA-Z']+", text.lower())
    return collections.Counter(words)


if __name__ == "__main__":
    sample = "the quick brown fox jumps over the lazy dog the fox runs"
    for word, count in word_count(sample).most_common(3):
        print(word, count)

File: training-data/forth/word_count.fs
: COUNT-WORDS ( addr len -- count )
  0 -ROT
  OVER + SWAP
  FALSE -ROT
  BEGIN
    2DUP <
  WHILE
    DUP C@ BL =
    IF
      DROP SWAP
    ELSE
      SWAP 0= IF ROT 1+ ROT SWAP THEN TRUE SWAP
    THEN
    ROT ROT
    1+ SWAP
  REPEAT
  2DROP DROP ;

S" the quick brown fox" COUNT-WORDS .
CR
